From 5c4cd3cee03428636e8d7cc4ed644389a4d598b3 Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Thu, 28 Apr 2011 02:56:42 -0500
Subject: [PATCH] posixtz: ensure the file offset we pass to lseek is off_t

on 32-bit systems, sizeof(off_t) is 4, on 64-bit sizeof(off_t) is 8
causing a word masking issue.
---
 posixtz.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/posixtz.c b/posixtz.c
index cddcb3e..972ca31 100644
--- a/posixtz-0.5/posixtz.c
+++ b/posixtz-0.5/posixtz.c
@@ -36,7 +36,7 @@ char *posix_tz(const char *filename)
 	if (r != TZ_BUFLEN
 	    || strncmp(buf, "TZif", 4) != 0
 	    || (unsigned char)buf[4] < 2
-	    || lseek(fd, -TZ_BUFLEN, SEEK_END) < 0
+	    || lseek(fd, (off_t) -TZ_BUFLEN, SEEK_END) < 0
 	   )
 		goto ERROR;
 
-- 
1.7.4.5

